### ISSUE TITLE
Shorten import statements

### ISSUE THREAD

**Issue by Evertt** [huntabyte/shadcn-svelte#114]
### Describe the feature in detail (code, mocks, or screenshots encouraged)

When I see this docs page:
https://www.shadcn-svelte.com/docs/components/alert-dialog

And I see an import statement like this:

```javascript
import {
    AlertDialog,
    AlertDialogAction,
    AlertDialogCancel,
    AlertDialogContent,
    AlertDialogDescription,
    AlertDialogFooter,
    AlertDialogHeader,
    AlertDialogTitle,
    AlertDialogTrigger
  } from "$components/ui/alert-dialog";
```

And that's just one UI component,  essentially. Once you have a full web page going on, your import statements are gonna be massive.

But it could be massively simplified. You could export the different parts in kind of nested object form.

Then the code would become this:
```html
<script lang="ts">
  import { Alert } from "$components/ui/alert-dialog";
  import { Button } from "$components/ui/button";
</script>

<!-- Notice the periods -->
<Alert.Dialog>
  <Alert.Dialog.Trigger>
    <Button variant="outline">Show Dialog</Button>
  </Alert.Dialog.Trigger>
  <Alert.Dialog.Content>
    <Alert.Dialog.Header>
      <Alert.Dialog.Title>Are you absolutely sure?</AlertDialogTitle>
      <Alert.Dialog.Description>
        This action cannot be undone. This will permanently delete your account
        and remove your data from our servers.
      </Alert.Dialog.Description>
    </Alert.Dialog.Header>
    <Alert.Dialog.Footer>
      <Alert.Dialog.Cancel>Cancel</Alert.Dialog.Cancel>
      <Alert.Dialog.Action>Continue</Alert.Dialog.Action>
    </Alert.Dialog.Footer>
  </Alert.Dialog.Content>
</Alert.Dialog>
```

I'm not actually sure if nesting more than 1 layer deep would actually work. But I did test and see that it works with 1 layer of nesting. So then you could still have just the simple `import { Alert } from "$components/ui/alert-dialog"` statement.

Or, if you want to give people the option to either import every sub-component or not, for tree-shaking purposes maybe, you could still export things in a way that people like me can import like so: `import * as Alert from "$components/ui/alert-dialog"`. By outputting the sub-components under two names, one with the `Alert` prefix, and one without that prefix.

I don't know, I'd like to hear what you guys think about this.

### What type of pull request would this be?

Other

**Comment by huntabyte** [comment]
I wanted to get confirmation that there weren't any concerns with tree-shaking before doing so, which one of the maintainers was able to give me clarification on so this will be coming soon!

**Comment by Evertt** [comment]
This could be another way of doing it. Just so that you have more options to think about:

https://www.youtube.com/watch?v=ZOBwBPt7q8Y

**Comment by huntabyte** [comment]
Closed by #150
